core/src/main/kotlin/transformers/descriptors/DefaultDescriptorToDocumentationTranslator.kt

```diff
@@ -70,6 +81,8 @@ class DokkaDescriptorVisitor(
             listOfNotNull(descriptorData),
             mutableSetOf() // TODO Implement following method to return proper results getXMLDRIs(descriptor, descriptorData).toMutableSet()
         )
+
+        return enum_values?.let { Enum(clazz, it) } ?: clazz
```

Comment:
Flow of logic in this function is really spaghettified. To make it more readable make this function check class kind, then dispatch to `visitEnumClassDescriptor` and `visitOrdinaryClassDescriptor` or something similarly named. If there is such need you can extract common logic.

Reply:
Bump

Reply:
Changed, should be simpler now.